src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -758,9 +758,20 @@ static int rbd_nbd(int argc, const char *argv[])
 
   argv_to_vec(argc, argv, args);
   env_to_vec(args);
-  auto cct = global_init(NULL, args, CEPH_ENTITY_TYPE_CLIENT,
-			 CODE_ENVIRONMENT_DAEMON,
-			 CINIT_FLAG_UNPRIVILEGED_DAEMON_DEFAULTS);
+
+  code_environment_t code_env = CODE_ENVIRONMENT_UTILITY;
+  int env_flag = 0;
+  for (auto arg : args) {
+    if (arg != NULL && arg[0] != '-') {
+      if (strcmp(arg, "map") == 0) {
+        code_env = CODE_ENVIRONMENT_DAEMON;
+        env_flag = CINIT_FLAG_UNPRIVILEGED_DAEMON_DEFAULTS;
+      }
+      break;
+    }
+  }
+
+  auto cct = global_init(NULL, args, CEPH_ENTITY_TYPE_CLIENT, code_env, env_flag);
```

Comment:
@trociny Are you only seeing this because you have a vstart cluster and the "[client]" section has entries for "log file" and "admin socket"? If that is the case and it works as expected when there are no config settings, I'm fine w/ the behavior as it relates to this tracker ticket. I'd still love to see a pluggable metavariable system where apps could define custom metavariables like "$device", but that is for another PR.